### PR TITLE
Neurosky source module

### PR DESCRIPTION
**Pull request by marionleborgne**

### CODE REVIEW

**Comment by marionleborgne** [comment]
If you want to try it, checkout this branch and:

### 1. Install cloudbrain
```
cd cloudbrain/
pip install -e . --user
```

### 2. Run the NeuroSky source module
> WARNING: you need to have a local RabbitMQ running since the module publishes to RMQ
```
python -m cloudbrain.run --file examples/source.neurosky.json
```
### 3. Alternatively you can directly run the connector (the thing that _accesses_ the neurosky packets)
```
python src/cloudbrain/connectors/neurosky.py
```

File: src/cloudbrain/connectors/neurosky.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+__changelog__ = """Last Update: 2017.05.28"""
+import threading
+import signal
+import sys
+
+from cloudbrain.connectors.thinkgear import SerialDevice
+from cloudbrain.connectors.thinkgear import puzzlebox_synapse_protocol_thinkgear
+
+THINKGEAR_DEVICE_SERIAL_PORT = '/dev/tty.MindWaveMobile-DevA'
```
Comment (flysonic10) [inline src/cloudbrain/connectors/neurosky.py]: Nevermind, I see that this is only the default and it's being overridden by the source module.

File: src/cloudbrain/connectors/neurosky.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+def callback_factory(metric_name):
+    def print_callback(timestamp, sample):
+        print 'metric_name=%s, timestamp=%s, sample=%s' % (metric_name,
+                                                           timestamp, sample)
+
+    return print_callback
+
+
+def example():
```
Comment (flysonic10) [inline src/cloudbrain/connectors/neurosky.py]: Nevermind for this comment, too. I see what's going on here. The other connectors don't have a main entry example. Should we just get rid of this?

File: examples/source.neurosky.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          "name": "PikaPublisher",
+          "package": "cloudbrain.publishers.rabbitmq",
+          "options": {
+            "rabbitmq_address": "localhost",
+            "rabbitmq_user": "guest",
+            "rabbitmq_pwd": "guest",
+            "rabbitmq_vhost": "/"
+          },
+          "base_routing_key": "some_unique_key",
+          "metrics": [
```
Comment (marionleborgne) [inline examples/source.neurosky.json]: Added

File: src/cloudbrain/connectors/neurosky.py (hunk 2 above)
Comment (marionleborgne) [inline src/cloudbrain/connectors/neurosky.py]: There is a ticket to add examples for all connectors so that they can be run in standalone mode.

**Comment by marionleborgne** [comment]
Thank you @castellotti for the help.

**Pull request merged**